gips/data/landsat/landsat.py

```diff
@@ -1001,20 +1016,43 @@ def process(self, products=None, overwrite=False, **kwargs):
 
                     elif val[0] == 'cloudmask':
                         qaimg = self._readqa()
-                        npqa = qaimg.Read() # transmogrify into numpy array
+                        npqa = qaimg.Read()  # read image file into numpy array
                         # https://landsat.usgs.gov/collectionqualityband
                         # cloudmaskmask = (cloud and (cc_med or cc_high)) or csc_med or csc_high
                         # cloud iff bit 4
                         # (cc_med or cc_high) iff bit 6
                         # (csc_med or csc_high) iff bit 8
+
                         def get_bit(np_array, i):
                             """Return an array with the ith bit extracted from each cell."""
                             return (np_array >> i) & 0b1
-                        np_cloudmask = get_bit(npqa, 4) & get_bit(npqa, 6) | get_bit(npqa, 8)
-                        imgout = gippy.GeoImage(fname, img, gippy.GDT_Byte, 1) # only one layer
+
+                        np_cloudmask = numpy.logical_not(
+                            get_bit(npqa, 4) &
+                            get_bit(npqa, 6) |
+                            get_bit(npqa, 8)
+                        )
+
+                        # We already have an implicit dependency on scipy (from
+                        # Py6S), but life might be more simple if we just move
+                        # the whole cloudmask extraction into gippy.
+                        from scipy import ndimage
+                        np_cloudmask_erroded = ndimage.binary_erosion(
+                            np_cloudmask,
+                            structure=numpy.ones((10, 10), dtype=numpy.uint8),
+                        )
+                        # 
+
+                        imgout = gippy.GeoImage(fname, img, gippy.GDT_Byte, 1)
                         verbose_out("writing " + fname, 2)
-                        imgout.SetBandName('Cloud Mask', 1)
-                        imgout[0].Write(np_cloudmask)
+                        imgout.SetBandName(
+                            self._products[val[0]]['bands'][0], 1
+                        )
+                        imgout.SetMeta('GIPS_LANDSAT_VERSION', self.version)
```

Comment:
Optional:  Maybe not now, but it seems like this should be set for all products

Reply:
Agreed...it was a "while I'm here" fix.  We need a metadata standardization effort.